packages/chat-core/src/index.ts

```diff
@@ -961,7 +961,7 @@ export function promisify<T>(
         _responses: Response[],
         newResponse: Response | undefined,
       ): void => {
-        if (newResponse?.type === "bot") {
+        if (newResponse?.type === "bot" || newResponse?.type === "failure") {
           clearTimeout(timeoutId);
           convo.unsubscribe(subscription);
           resolve(newResponse);
```

Comment:
should it resolve or fail here?

Reply:
I don't think the error can be typed accurately, so my take is that unless we can do something about that, definitely no rejection.

Reply:
yeah, trade-offs. 

On the one hand: better typechecking through the whole system, but two ways to have errors

On the other hand: a single way to detect errors that's enforced through the language, but forcing type-checking to, e.g. check if the error object has a `response` field or something.

Reply:
I'm not exactly married to my opinion above especially since failure could just reject with a string (that's pretty much what it is), the road I don't feel good about walking down on is more like the philosophical rabbit hole of defining what an error is. There are cases where bot-type responses technically mean failure (it is in the message metadata), which would end up in the `.then` branch at the moment, and it seems very cumbersome to separate it out of there.

Reply:
I will merge this PR now, @michaelglass and @jakub-nlx feel free to suggest alternative API's at any time with this discussion in mind.